Epsilon/Services/FilterService.cs

```diff
@@ -59,18 +59,30 @@ public async Task<IEnumerable<EnrollmentTerm>> GetParticipatedTerms(string stude
         {
             return Enumerable.Empty<EnrollmentTerm>();
         }
-
+        
         var submissions = response.Courses!.SelectMany(static c => c.Submissions!.Nodes);
 
         var participatedTerms = allTerms!
-                                .Where(static term => term is { StartAt: not null, EndAt: not null, })
-                                .Where(term => submissions.Any(submission => submission.SubmittedAt >= term.StartAt && submission.SubmittedAt <= term.EndAt))
-                                .Distinct()
-                                .OrderByDescending(static term => term.StartAt);
-
-        return participatedTerms;
+            .Where(static term => term is { StartAt: not null, EndAt: not null, })
+            .Where(term => submissions.Any(submission => submission.SubmittedAt >= term.StartAt && submission.SubmittedAt <= term.EndAt))
+            .Distinct()
+            .OrderByDescending(static term => term.StartAt).ToList();
+        
+        // Get the corrected term based on a new end date:
+        var correctedParticipatedTerms = participatedTerms
+            .Select((currentTerm, index) => new EnrollmentTerm(
```

Comment:
Use 'with' expression to copy record. 
Please use Qodana suggestion

Reply:
Great, I pushed the new version with the expression implemented